src/django/api/views/v1/production_locations.py

```diff
@@ -184,3 +188,58 @@ def create(self, request):
             },
             status=result.status_code
         )
+
+    @transaction.atomic
+    def partial_update(self, request, pk=None):
+        if switch_is_active('disable_list_uploading'):
+            raise ServiceUnavailableException(
+                APIV1CommonErrorMessages.MAINTENANCE_MODE
+            )
+        if not Facility.objects.filter(id=pk).exists():
+            specific_error = APIV1LocationContributionErrorMessages \
+                .LOCATION_NOT_FOUND
+            return Response(
+                {'detail': specific_error},
+                status=status.HTTP_404_NOT_FOUND
+            )
+        if not isinstance(request.data, dict):
```

Comment:
**Suggestion:** Move it ti ProductionLocationService as was implemented in https://github.com/opensupplyhub/open-supply-hub/pull/438 (ModerationEventService)

Reply:
If you don't mind, I would prefer to leave it as is, because it’s a simple check that I don’t think requires a special method.